src/signal.rs

```diff
@@ -156,6 +156,13 @@ pub fn validate_signal_num(num: c_int) -> errno::Result<()> {
 pub fn register_signal_handler(num: c_int, handler: SignalHandler) -> errno::Result<()> {
     validate_signal_num(num)?;
 
+    // signum specifies the signal and can be any valid signal except
```

Comment:
Should this check be in validate_signal_num as well?

Reply:
my opinion is not needed because _validate_signal_num_ is generic util and this restriction only applies specific to _register_signal_handler_ alone 🙂

Reply:
Ok.

Reply:
@andreeaflorescu I have covered unit test 🙂